packages/clay-core/src/tree-view/useMultipleSelection.tsx

```diff
@@ -23,22 +24,34 @@ export interface IMultipleSelection {
 }
 
 export interface IMultipleSelectionState {
-	createPartialLayoutItem: (key: Key, parentKey?: Key) => () => void;
+	createPartialLayoutItem: (
+		key: Key,
+		lazy: boolean,
+		loc: Array<number>,
+		parentKey?: Key
+	) => () => void;
 	isIntermediate: (key: Key) => boolean;
 	selectedKeys: Set<Key>;
 	toggleSelection: (key: Key) => void;
 }
 
 export interface IMultipleSelectionProps<T>
 	extends IMultipleSelection,
-		ITreeProps<T>,
+		Pick<ITreeProps<T>, 'nestedKey'>,
 		Pick<ICollectionProps<T>, 'items'> {
 	selectionMode?: 'multiple' | 'single';
 }
 
 type LayoutInfo = {
 	children: Set<Key>;
 	intermediate: boolean;
+
+	/**
+	 * Lazy Child means that the current Node has children but they were not
+	 * created because they are not visible in the DOM.
+	 */
+	lazyChild: boolean;
+	loc: Array<number>;
```

Comment:
Hey @matuzalemsteles, I'm just reviewing this change in GitHub, and I was wondering, what does `loc` mean here?

Reply:
Oh sorry @julien, this is actually `indexes`, just short for localization 😅, we can switch to indexes anyway, it's the custom.

Reply:
Oh I see, if it's `indexes`, then that makes sense. Thanks for clarifying.